src/flamenco/types/gen_stubs.py

```diff
@@ -1358,13 +1594,60 @@ def emitWalk(self, inner):
                 print(f'    {namespace}_{self.element}_walk( w, self->{self.name}, fun, "{self.name}", level );', file=body)
             print( '  }', file=body)
 
+    def emitDecodeArchival(self):
+        print('  {', file=body)
+        print('    uchar o;', file=body)
+        print('    fd_bincode_bool_decode_unsafe( &o, ctx );', file=body)
+        if self.flat:
+            print(f'    self->has_{self.name} = !!o;', file=body)
+            print('    if( o ) {', file=body)
+            if self.element in simpletypes:
+                print(f'      fd_bincode_{simpletypes[self.element]}_decode( &self->{self.name}, ctx );', file=body)
+            else:
+                el = f'{namespace}_{self.element}'
+                el = el.upper()
+                print(f'      {namespace}_{self.element}_new( &self->{self.name} );', file=body)
+                print(f'      {namespace}_{self.element}_decode_archival( &self->{self.name}, ctx );', file=body)
+            print('    }', file=body)
+        else:
+            print('    if( o ) {', file=body)
+            if self.element in simpletypes:
+                print(f'      self->{self.name} = fd_valloc_malloc( ctx->valloc, 8, sizeof({self.element}) );', file=body)
```

Comment:
Should we use an alignment of less than 8 for those simple types that have an alignment of less than 8?

Reply:
fd_alloc will never do alignment of less than 8 anyway